### DISCUSSION TITLE
Koordinaten

### DISCUSSION THREAD

**Discussion by Walterbraunlage** [Schrolli91/BOSWatch#547] [General]
Nabend,
ich hab ne  koordinaten frage.
in der config kann man die koordinaten anzeigen lassen.

```
# With some message, coordinates can be sent to determine the destination
# Do you want to enable this feature? (0 - off | 1 - on)
geo_enable = 0
# If a RIC contains coordinates, specify the regex used to decode them
geo_format = #C(\d{2})(\d{5}),(\d{2})(\d{5})#
geo_order = LON, lon, LAT, lat

```
wenn meine nachricht so aussieht, was muss ich dort eingeben?
`#K01;N5330111E5322111;Bundesautobahn A1 BAB1 > Köln/O  AS Bonn West - AS Bonn O 23    H:Hilfeleistung    6854 03:34`

und wenn ich ne weiterleitung zu bosmon, habe kommt das da an?
oder was wird mit der erkenntniss gemacht?

vielen dank

**Comment by Schrolli91** [comment]
Zum probieren mit Regex empfehle ich immer https://regexr.com/
Hier mal als Beispiel - must du ggf. noch auf deinen Fall etwas Anpassen: https://regexr.com/5lk8s

Das Bosmon Plugin macht damit aber, soweit mir bekannt, gar nichts, da die Bosmon API keine Koordinaten entgegen nimmt.
Zumindest nicht in der aktuell Plugin Implementierung.
Bei Telegram würde aber zB. der Standort mit versendet.

**Comment by Skipper1990** [comment]
Morgen,

geo_format = #C(\d{2})(\d{5}),(\d{2})(\d{5})#

das #C am Anfang und die # am Ende muss so bleiben ? Oder muss in dem Fall K01; anstatt C dahin ?

**Comment by Skipper1990** [comment]
2te frage.
Wenn ich es aktiviert haben, brauch ich da eine GoogleAPIKey oder geht das so über Telegramm.
Im Moment hat es nicht funktioniert

**Comment by Schrolli91** [comment]
Nein das #C am Anfang sollte nicht notwendig sein, außer in deinem Text steht #C zu Beginn.
Das von mir gepostete Beispiel sollte doch schon recht gut zu deinem Beispiel passen.
Probiere es damit einfach mal aus. Sollte es Probleme geben bitte direkt mit Logauszug.

**Comment by Skipper1990** [comment]
Folgender fehler kommt im Log:

10.02.2021 13:03:59 - decoder         [DEBUG   ] received POCSAG
10.02.2021 13:03:59 - poc             [DEBUG   ] POCSAG Bitrate: 1200
10.02.2021 13:03:59 - poc             [DEBUG   ] Using #N(\d{2})(\d+)E(\d{2})(\d+)# to find geo-tag in #K01;N51xxxxxxE13xxxxx; -mSoSi -xxxxxxxxxxxxxxxxxx
10.02.2021 13:03:59 - poc             [DEBUG   ] No geo-tag found



So sieht der Filter aktuell aus.
geo_format = #N(\d{2})(\d+)E(\d{2})(\d+)#

**Comment by MarSch1** [comment]
#N kommt ja in Deinem POC Text gar nicht vor, sondern #K01;N.

Probier mal den Filter:
;N(\d{2})(\d+)E(\d{2})(\d+);

**Comment by Skipper1990** [comment]
Habe ich mal so geändert.
Wenn dies auch nicht geht versuche ich mal #K01;N.......

**Comment by Schrolli91** [comment]
Auch der Hastag am Ende ist ja falsch - versuch doch einfach den Regex, welchen ich im Beispiel verwendet habe, der sollte eigentlich matchen => `N(\d{2})(\d+)E(\d{2})(\d+)`
Das so eine Kombination irgendwo im Text vorkommt ist ja eher unwahrscheinlich, denke ich

**Comment by Skipper1990** [comment]
Jetzt hat der Filter funktioniert.


10.02.2021 17:17:28 - decoder         [DEBUG   ] received POCSAG
10.02.2021 17:17:28 - poc             [DEBUG   ] POCSAG Bitrate: 1200
10.02.2021 17:17:28 - poc             [DEBUG   ] Using N(\d{2})(\d+)E(\d{2})(\d+) to find geo-tag in #K01;N51xxxxxxE13xxxxxx; msosi xxxxxxx
10.02.2021 17:17:28 - poc             [DEBUG   ] Found geo-tag in message, parsing...
10.02.2021 17:17:28 - poc             [DEBUG   ] Finished parsing geo; lon: 51.xxxxxx, lat: 13.xxxxxx


Bloß war jetzt keine aktion bei Telegramm, brauche ich eine google Apikey oder was muss ich da aktivieren?

**Comment by flothi** [comment]
Ja, das Plugin in seiner jetzigen Form ist hart mit googleAPI verheiratet; aber Telegram kann das meines Wissens nach auch so. So, wie das Plugin hier geschrieben ist, werden zwei Bilder angezeigt (Übersicht und Detail)
